### DISCUSSION TITLE
Include a .deb package for releases

### DISCUSSION THREAD

**Discussion by JacobCoffee** [astral-sh/ruff#3429] [Ideas]
[`cargo-deb`](https://crates.io/crates/cargo-deb) is an add-on to the existing Cargo tooling that allows you to output your project to a `.deb` binary.

These are used within the Debian/Ubuntu ecosystem.

Would it be possible to add this on to our release process for end users to download and install with `dpkg`, `apt`, etc. ?

If so, and I am probably wrong in the scope of changes needed but, I believe there are some additions to the `Cargo.toml` that need to be made to adhere to [Debian's `control` file conventions](https://www.debian.org/doc/debian-policy/ch-controlfields.html#list-of-fields).

**Comment by charliermarsh** [comment]
\cc @konstin - any thoughts here as the local packaging expert? :)

**Reply by konstin** [reply]
[cargo-dev](https://github.com/kornelski/cargo-deb) should be able to do this

**Reply by charliermarsh** [reply]
Do we have to build separate `.deb` binaries for different architectures?

**Reply by konstin** [reply]
yes, see e.g. https://packages.ubuntu.com/focal/utils/ripgrep

**Reply by JacobCoffee** [reply]
I am interested in helping with this one if it is something you choose to pursue, though I admit it may be like an ant trying to lift a shoe in the amount I can get done.

I have used `cargo-deb` before though, and it was pretty straightforward.
